libportability-gfx/src/impls.rs

```diff
@@ -949,14 +952,345 @@ pub extern "C" fn gfxMergePipelineCaches(
+
+            (bindings, attributes)
+        };
+
+        let input_assembler = {
+            let input_state = unsafe { &*info.pInputAssemblyState };
+            let tessellation_state = if shaders.hull.is_some() {
+                Some(unsafe { &*info.pTessellationState })
+            } else {
+                None
+            };
+
+            assert_eq!(input_state.primitiveRestartEnable, VK_FALSE); // TODO
+
+            pso::InputAssemblerDesc {
+                primitive: conv::map_primitive_topology(
+                    input_state.topology,
+                    tessellation_state
+                        .map(|state| state.patchControlPoints as _)
+                        .unwrap_or(0),
+                ),
+                primitive_restart: pso::PrimitiveRestart::Disabled, // TODO
+            }
+        };
+
+        // TODO: `pColorBlendState` could contain garbage, but implementations
+        //        can ignore it in some circumstances. How to handle it?
+        let blender = {
+            let mut blend_desc = pso::BlendDesc::default();
+
+            let blend_state = unsafe { info.pColorBlendState.as_ref() };
+            if let Some(state) = blend_state {
+                if state.logicOpEnable == VK_TRUE {
+                    blend_desc.logic_op = Some(conv::map_logic_op(state.logicOp));
+                }
+
+                let attachments = unsafe {
+                    slice::from_raw_parts(state.pAttachments, state.attachmentCount as _)
+                };
+                blend_desc.targets = attachments
+                    .into_iter()
+                    .map(|attachment| {
+                        let color_mask = conv::map_color_components(attachment.colorWriteMask);
+
+                        let blend = if attachment.blendEnable == VK_TRUE {
+                            pso::BlendState::On {
+                                color: conv::map_blend_op(
+                                    attachment.colorBlendOp,
+                                    attachment.srcColorBlendFactor,
+                                    attachment.dstColorBlendFactor,
+                                ),
+                                alpha: conv::map_blend_op(
+                                    attachment.alphaBlendOp,
+                                    attachment.srcAlphaBlendFactor,
+                                    attachment.dstAlphaBlendFactor,
+                                ),
+                            }
+                        } else {
+                            pso::BlendState::Off
+                        };
+
+                        pso::ColorBlendDesc(color_mask, blend)
+                    })
+                    .collect();
+            }
+
+            // TODO: multisampling
+
+            blend_desc
+        };
+
+        // TODO: `pDepthStencilState` could contain garbage, but implementations
+        //        can ignore it in some circumstances. How to handle it?
```

Comment:
what are those circumstances/cases?

Reply:
If the subpass doesn't use a depth-stencil attachment or rasterization is disabled (this part we can check) `pDepthStencilState` is ignored. The specification doesn't explicitly enforce that it must be NULL.